vars/daosLatestVersion.groovy

```diff
@@ -23,13 +23,12 @@ String call(String next_version='1000', String repo_type='stable') {
                 case 'weekly-testing':
                     _next_version = 1000
                     break
-                case 'release/2.2':
-                case 'weekly-testing-2.2':
-                    _next_version = 2.3
-                    break
-                case 'release/2.0':
-                case 'weekly-testing-2.0':
-                    _next_version = 2.1
+                case ~/^(release|weekly).*(\/|-)(\d+\.\d+)/:
+                    version = Matcher.lastMatcher.group(3)
+                    major = version.split('.')[0] as Integer
+                    minor = version.split('.')[1] as Integer
```

Comment:
Why not use regex for all of the lifting?

```suggestion
                case ~/^(release|weekly).*(\/|-)(\d+)\.(\d+)/:
                    version = 
                    major =  Matcher.lastMatcher.group(3) as Integer
                    minor =  Matcher.lastMatcher.group(4) as Integer
```

Reply:
Seems reasonable.